internal/hcp/internal/controllers/telemetrystate/controller.go

```diff
@@ -153,16 +141,63 @@ func ensureTelemetryStateDeleted(ctx context.Context, rt controller.Runtime) err
 	return nil
 }
 
-// getLinkResource returns the cluster scoped pbhcp.Link resource. If the resource is not found a nil
-// pointer and no error will be returned.
-func getLinkResource(ctx context.Context, rt controller.Runtime) (*types.DecodedLink, error) {
-	resp, err := rt.Client.Read(ctx, &pbresource.ReadRequest{Id: &pbresource.ID{Name: "global", Type: pbhcp.LinkType}})
+func writeTelemetryStateIfUpdated(ctx context.Context, rt controller.Runtime, state *pbhcp.TelemetryState) error {
+	currentState, err := getTelemetryStateResource(ctx, rt)
+	if err != nil {
+		return err
+	}
+
+	if currentState != nil && proto.Equal(currentState, state) {
```

Comment:
I think we need `GetData()` in order to get at the `*TelemetryState`

```suggestion
	if currentState != nil && proto.Equal(currentState.GetData(), state) {
```